src/StackExchange.Redis/Message.cs

```diff
@@ -222,6 +227,10 @@ internal void SetScriptUnavailable()
 
         public bool IsFireAndForget => (Flags & CommandFlags.FireAndForget) != 0;
         public bool IsInternalCall => (Flags & InternalCallFlag) != 0;
+        public bool IsOnConnectionRestoreRetryIfNotYetSent => (Flags & CommandFlags.OnConnectionRestoreRetryIfNotYetSent) != 0;
+        public bool IsOnConnectionRestoreAlwaysRetry => (Flags & CommandFlags.OnConnectionRestoreAlwaysRetry) != 0;
+        public bool IsOnConnectionRestoreNoRetry => (Flags & CommandFlags.OnConnectionRestoreNoRetry) != 0;
+        private bool IsOnConnectionRestoreFlagSet => IsOnConnectionRestoreAlwaysRetry || IsOnConnectionRestoreRetryIfNotYetSent || IsOnConnectionRestoreNoRetry;
```

Comment:
This approach adds `n` properties for `n` retry notions, let's chat and see if we can remove that cardinality on the central Message component.

Reply:
After unifying config option command retry with command flags this is improved.